SQLAssignment.txt

```diff
@@ -0,0 +1,1345 @@
+| 3005 | 5160.45 | 1990-03-10 | 2003 | 1002 |
+| 3006 | 1098.16 | 1990-03-10 | 2008 | 1007 |
+| 3009 | 1713.23 | 1990-04-10 | 2002 | 1003 |
++------+---------+------------+------+------+
+
+                                                              ALL QUESTIONS AND THEIR QUERIES :- 
+
+1) List all the columns of the Salespeople table.
+-> select * from salespeople;
+
++------+---------+-----------+------+
+| SNUM | SNAME   | CITY      | COMM |
++------+---------+-----------+------+
+| 1001 | Peel    | London    |    0 |
+| 1002 | Serres  | SanJose   |    0 |
+| 1003 | AxelRod | New York  |    0 |
+| 1004 | Motika  | London    |    0 |
+| 1007 | Rifkn   | Barcelona |    0 |
+| 1008 | Fran    | London    |    0 |
++------+---------+-----------+------+
+
+2) List all customers with a rating of 100.
+-> SELECT * from customers WHERE RATING = 100;
+
++------+---------+--------+--------+------+
+| CNUM | CNAME   | CITY   | RATING | SNUM |
++------+---------+--------+--------+------+
+| 2001 | Hoffman | London |    100 | 1001 |
+| 2005 | Clemens | London |    100 | 1001 |
+| 2007 | Pereira | Rome   |    100 | 1004 |
++------+---------+--------+--------+------+
+
+3) Find the largest order taken by each salesperson on each date.
+-> SELECT o.ONUM,MAX(AMT) ,ODATE,c.CNUM FROM orders o, customers c WHERE o.CNUM=c.CNUM GROUP BY ODATE,c.SNUM,o.ODATE;
+
++------+----------+------------+------+
+| ONUM | MAX(AMT) | ODATE      | CNUM |
++------+----------+------------+------+
+| 3003 |   767.19 | 1990-10-03 | 2001 |
+| 3005 |  5160.45 | 1990-10-03 | 2003 |
+| 3009 |  1713.23 | 1990-10-03 | 2002 |
+| 3002 |  1900.10 | 1990-10-03 | 2007 |
+| 3008 |  9891.88 | 1990-10-03 | 2006 |
++------+----------+------------+------+
+
+4) Arrange the Order table by descending customer number.
+-> SELECT * from orders Order by CNUM DESC;
+
++------+---------+------------+------+------+
+| ONUM | AMT     | ODATE      | CNUM | SNUM |
++------+---------+------------+------+------+
+| 3001 |   18.69 | 1990-10-03 | 2008 | 1007 |
+| 3006 | 1098.16 | 1990-10-03 | 2008 | 1007 |
+| 3002 | 1900.10 | 1990-10-03 | 2007 | 1004 |
+| 3008 | 4723.00 | 1990-10-03 | 2006 | 1001 |
+| 3011 | 9891.88 | 1990-10-03 | 2006 | 1001 |
+| 3007 |   75.75 | 1990-10-03 | 2004 | 1002 |
+| 3010 | 1309.95 | 1990-10-03 | 2004 | 1002 |
+| 3005 | 5160.45 | 1990-10-03 | 2003 | 1002 |
+| 3009 | 1713.23 | 1990-10-03 | 2002 | 1003 |
+| 3003 |  767.19 | 1990-10-03 | 2001 | 1001 |
++------+---------+------------+------+------+
+
+5) Find which salespeople currently have orders in the order table.
+-> SELECT SNAME from salespeople,orders  where orders.SNUM=salespeople.SNUM;
+
++---------+
+| SNAME   |
++---------+
+| Peel    |
+| Peel    |
+| Peel    |
+| Serres  |
+| Serres  |
+| Serres  |
+| AxelRod |
+| Motika  |
+| Rifkn   |
+| Rifkn   |
++---------+
+
+6) List names of all customers matched with the salespeople serving them.
+-> SELECT SNAME  from salespeople,customers  where Customers.SNUM=salespeople.SNUM;
```

Comment:
Here we are expecting the customer's name but you are showing salesperson's name.